compiler/rustc_trait_selection/src/traits/select/mod.rs

```diff
@@ -2465,7 +2488,14 @@ struct ProvisionalEvaluationCache<'tcx> {
     ///   means the cached value for `F`.
     map: RefCell<FxHashMap<ty::PolyTraitPredicate<'tcx>, ProvisionalEvaluation>>,
 
-    wf_tys: RefCell<Vec<(ty::GenericArg<'tcx>, usize)>>,
+    /// The stack of args that we assume to be true because a `WF(arg)` predicate
+    /// is on the stack above (and because of wellformedness is coinductive).
+    /// In an "ideal" world, this would share a stack with trait predicates in
+    /// `TraitObligationStack`. However, trait predicates are *much* hotter than
+    /// `WellFormed` predicates, and it's very likely that the additional matches
+    /// will have a perf effect. The value here is the well-formed `GenericArg`
```

Comment:
Maybe run perf on that first so we can replace the `very likely` with `been shown`

Reply:
Ehh, I would prefer to not try to make that change right now (because it's going to take some time to do right - which is limited for me right now - and I think the time will just be wasted).